src/storage/s3/storage.c

```diff
@@ -206,8 +208,9 @@ storageS3Auth(
             const String *headerKey = strLstGet(headerList, headerIdx);
             const String *headerKeyLower = strLower(strDup(headerKey));
 
-            // Skip the authorization header -- if it exists this is a retry
-            if (strEq(headerKeyLower, HTTP_HEADER_AUTHORIZATION_STR))
+            // Skip headers that do not require signing
+            if (!strEq(headerKeyLower, HTTP_HEADER_HOST_STR) && !strEq(headerKeyLower, HTTP_HEADER_CONTENT_MD5_STR) &&
+                !strBeginsWith(headerKeyLower, S3_HEADER_PREFIX_STR))
```

Comment:
I'm a little concerned that the code change, which does more than the ask, may result in issues that we are unaware of. Maybe you could point me to where the S3 documentation indicates that only headers that begin with "x-amz-" or are "host" or  "content-md5" are the only things that need to be signed?

Reply:
It's here: https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html. Search for `SignedHeaders`.  content-md5 is not in the list but that's not the only place where there docs are inaccurate.

Since content-length is the only thing we don't need to sign and we are not likely to add new headers, I would be OK with flipping this and just excluding content-length and authorization from signing.

Reply:
Done in cf8f611ac6dbd4c.

Reply:
Super, thanks.